cibuildwheel/macos.py

```diff
@@ -641,11 +640,18 @@ def build(options: Options, tmp_path: Path) -> None:
 
             # we're all done here; move it to output (overwrite existing)
             if compatible_wheel is None:
-                with contextlib.suppress(FileNotFoundError):
-                    (build_options.output_dir / repaired_wheel.name).unlink()
-
-                shutil.move(str(repaired_wheel), build_options.output_dir)
-                built_wheels.append(build_options.output_dir / repaired_wheel.name)
+                output_dir = build_options.output_dir.resolve()
+                output_wheel = output_dir.joinpath(repaired_wheel.name).resolve()
+                output_wheel.unlink(missing_ok=True)
+
+                # os.move() will rename the file to what we were expecting to be the parent directory if we don't ensure it exists
+                output_dir.mkdir(parents=True, exist_ok=True)
+
+                # using os.move() as Path.rename() is not guaranteed to work across filesystem boundaries
+                shutil.move(
+                    str(repaired_wheel), str(output_wheel)
+                )  # explicit str() needed for mypy3.8
```

Comment:
```suggestion
                )  # explicit str() needed for Python 3.8
```

https://docs.python.org/3/library/shutil.html#shutil.move